pyface/image/image.py

```diff
@@ -0,0 +1,1336 @@
+        license     = self.license.replace(     '\n', '\n    ' ).strip()
+        image_names = self.image_names
+        image_names.sort()
+        if len( image_names ) == 0:
+            image_names = [ 'All' ]
+        images = '\n'.join( [ '  - ' + image_name
+                              for image_name in image_names ] )
+
+        return (ImageVolumeInfoTextTemplate % ( description, self.copyright,
+                                                license, images ))
+
+    #-- Public Methods ---------------------------------------------------------
+
+    def clone ( self ):
+        """ Returns a copy of the ImageVolumeInfo object.
+        """
+        return self.__class__(**self.trait_get('description', 'copyright',
+                                               'license'))
+
+#-------------------------------------------------------------------------------
+#  'ImageVolume' class:
+#-------------------------------------------------------------------------------
+
+class ImageVolume ( HasPrivateTraits ):
+
+    # The canonical name of this volume:
+    name = Str
+
+    # The list of volume descriptors that apply to this volume:
+    info = List( ImageVolumeInfo )
+
+    # The category that the volume belongs to:
+    category = Str( 'General' )
+
+    # A list of keywords used to describe the volume:
+    keywords = List( Str )
+
+    # The list of aliases for this volume:
+    aliases = List( Str )
+
+    # The path of the file that defined this volume:
+    path = File
+
+    # Is the path a zip file?
+    is_zip_file = Bool( True )
+
+    # The FastZipFile object used to access the underlying zip file:
+    zip_file = Instance( FastZipFile )
+
+    # The list of images available in the volume:
+    images = List( ImageInfo )
+
+    # A dictionary mapping image names to ImageInfo objects:
+    catalog = Property( depends_on = 'images' )
+
+    # The time stamp of when the image library was last modified:
+    time_stamp = Str
+
+    # A read-only string containing the Python code needed to construct this
+    # ImageVolume object:
+    image_volume_code = Property
+
+    # A read-only string containing the Python code needed to construct the
+    # 'images' list for this ImageVolume object:
+    images_code = Property
+
+    # A read-only string containing the text describing the contents of the
+    # volume (description, copyright, license information, and the images they
+    # apply to):
+    license_text = Property
+
+    #-- Public Methods ---------------------------------------------------------
+
+    def update ( self ):
+        """ Updates the contents of the image volume from the underlying
+            image store, and saves the results.
+        """
+        # Unlink all our current images:
+        for image in self.images:
+            image.volume = None
+
+        # Make sure the images are up to date by deleting any current value:
+        del self.images
```

Comment:
I would think that `self.images = []`  or self.reset_traits(['images']) is better than removing the `images` attribute